hugo/content/blog/2021-12-07-new-versioning-scheme/index.md

```diff
@@ -0,0 +1,54 @@
+---
+title: New versioning scheme
+author: Robert Adam
+date: 2021-12-07
+categories:
+  - News
+---
+Mumble traditionally used a versioning scheme of the form major.minor.patch where the patch version number was incremented for every bug-fix release
+of a stable release series (e.g. 1.3.0 -> 1.3.1). Most notably though, the patch version was always starting at zero for a new stable release series
+(e.g. 1.2.0, 1.3.0, etc.). This will no longer be the case.
+
+<!--more-->
+
+## Background
+
+Mumble was always using the [semantic versioning scheme](https://semver.org/) which generally makes it really easy to know what to expect when
+switching from one version to another. While this is working great, using only the three common version numbers (major.minor.patch) poses a problem
+when considering snapshot builds. A snapshot build is essentially a beta-release for an upcoming release. Thus it has to use the same version numbers
+as that upcoming release. Thus, in order to get a unique versioning scheme, we had to make use of the fourth version number, that semantic versioning
+allows for. We called this the _build number_. This build number was just a number that was incremented with every build of the Mumble binaries that
+we performed. Thus a version could now look like this: 1.4.0.62 and the stable release would then simply use whatever build number was next in the
+queue.
+
+## The problem
+
+In theory that worked great and solved the issue at hand: we now had a unique version for every release (be that a stable or a snapshot release).
+However, as it turns out Windows hasn't arrived in the 21st century yet: it restricts programs to use a three-component versioning scheme. While it is
+possible to set a fourth component, without causing any issues (it is even displayed when checking the program's version), it is ignored for all
+relevant actions such as upgrading to a newer version. However, it seems it wasn't ignored entirely as using a four-component versioning scheme caused
+Windows to install a second (or third, or ...) version of Mumble alongside the already installed version if one was to attempt to upgrade between
+Mumble versions that only differed in the build number.
+
+## The solution
+
+After having researched quite a bit, we came to the conclusion that it seems to be impossible to get Windows to properly respect the build number in
+our version specifier (old dogs don't learn new tricks), so it was clear that we had to find a different solution to this. We evaluated a few
+different possibilities but eventually settled on using the patch version number as a replacement for the build number (which will be dropped). Thus,

```

Comment:
I think this is a bit confusing; I had to second-take because parentheses may grammatically refer to the patch version here.

```suggestion
different possibilities but eventually settled on using the patch version number as a replacement for the build number. The build number will be dropped. Thus,

```